src/core/mediator/UnitsHelper.cpp

```diff
@@ -12,27 +15,48 @@ double UnitsHelper::get_converted_value(const std::string &in_units, double valu
         throw std::runtime_error("Unable to process empty units value for pairing \"" + in_units + "\" \"" + out_units + "\"");
     }
     std::call_once(unit_system_inited, init_unit_system);
-    ut_unit* from = ut_parse(unit_system, in_units.c_str(), UT_UTF8);
-    if (from == NULL)
-    {
-        throw std::runtime_error("Unable to parse in_units value " + in_units);
-    }
-    ut_unit* to = ut_parse(unit_system, out_units.c_str(), UT_UTF8);
-    if (to == NULL)
-    {
-        ut_free(from);
-        throw std::runtime_error("Unable to parse out_units value " + out_units);
-    }
-    cv_converter* conv = ut_get_converter(from, to);
-    if (conv == NULL)
-    {
-        ut_free(from);
-        ut_free(to);
-        throw std::runtime_error("Unable to convert " + in_units + " to " + out_units);
-    }
-    double r = cv_convert_double(conv, value);
-    ut_free(from);
-    ut_free(to);
-    cv_free(conv);
+
+    auto converter = get_converter(in_units, out_units);
+    
+    double r = cv_convert_double(converter.get(), value);
     return r;
-}
\ No newline at end of file
+}
+
+std::shared_ptr<cv_converter> UnitsHelper::get_converter(const std::string& in_units, const std::string& out_units, utEncoding in_encoding, utEncoding out_encoding ){
+    const std::lock_guard<std::mutex> lock(converters_mutex);
+
+    std::string key = in_units + "|" + out_units; //Better solution? Good enough? Bother with nested maps?
```

Comment:
depends on what is "faster", string concat and then hash, or two hashes?

Reply:
Yeah, that, but I was actually thinking more about if `|` is somehow a valid character in a unit definition (probably not??) and/or if we could hit string encoding issues that this operator and std::string can't handle...no idea there.